app/controllers/career_site/positions_controller.rb

```diff
@@ -166,4 +169,17 @@ def set_cors_headers
     response.headers["Access-Control-Expose-Headers"] = "X-CSRF-Token"
     response.headers["Content-Security-Policy"] = "frame-ancestors *"
   end
+
+  def set_locale
+    locale = params[:locale]
+    if locale && I18n.available_locales.include?(locale.to_sym)
+      I18n.locale = locale
+      @locale = locale
+    elsif locale.present?
+      I18n.locale = I18n.default_locale
+      flash[:alert] = I18n.t("errors.locale_not_available", language: locale)
+    else
+      I18n.locale = I18n.default_locale
```

Comment:
Without `else`  branch I encountered a behavior that sometimes after using a locale in the parameters when reloading the page it loaded in a locale that was not passed as a parameter. This was not often. I did not understand the reason